homeassistant/components/google_generative_ai_conversation/__init__.py

```diff
@@ -67,7 +65,13 @@ async def generate_content(call: ServiceCall) -> ServiceResponse:
 
         prompt_parts = [call.data[CONF_PROMPT]]
 
-        def append_images_to_prompt():
+        config_entry: GoogleGenerativeAIConfigEntry = hass.config_entries.async_entries(
+            DOMAIN
+        )[0]
+
+        client = config_entry.runtime_data
+
+        def append_files_to_prompt():
             image_filenames = call.data[CONF_IMAGE_FILENAME]
             filenames = call.data[CONF_FILENAMES]
             for filename in image_filenames + filenames:
```

Comment:
Use a set to avoid duplicates? Until we remove image_filename I can see users getting confused and specifying the same file to both params which will result to duplicate uploads.

Reply:
Good idea; done